app/src/main/java/org/mozilla/focus/settings/permissions/AutoplayHelper.kt

```diff
@@ -0,0 +1,27 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.focus.settings.permissions
+
+import android.content.Context
+import org.mozilla.focus.R
+
+object AutoplayHelper {
+
+    fun getPermissionKeyIdByKeyValue(value: String, context: Context) = with(context) {
+        when (value) {
+            getString(R.string.pref_key_allow_audio_video_autoplay) -> R.string.pref_key_allow_audio_video_autoplay
+            getString(R.string.pref_key_block_autoplay_cellular_data) -> R.string.pref_key_block_autoplay_cellular_data
+            getString(R.string.pref_key_block_audio_video_autoplay) -> R.string.pref_key_block_audio_video_autoplay
+            else -> R.string.pref_key_block_autoplay_audio_only
+        }
+    }
+
+    fun getPermissionValueIdIdByKeyId(autoplayOptionId: Int) =
+        when (autoplayOptionId) {
+            R.string.pref_key_allow_audio_video_autoplay -> R.string.preference_allow_audio_video_autoplay
+            R.string.pref_key_block_autoplay_cellular_data -> R.string.preference_block_autoplay_cellular_data
+            R.string.pref_key_block_audio_video_autoplay -> R.string.preference_block_autoplay_audio_video
+            else -> R.string.preference_block_autoplay_audio_only
+        }
```

Comment:
I think you could remove the `getPermissionKeyIdByKeyValue` if you change  a bit the `getPermissionValueIdIdByKeyId` function.

```kotlin
    fun getPermissionValueIdIdByKeyId(value: String, context: Context) = with(context) {
        when (value) {
            getString(R.string.pref_key_allow_audio_video_autoplay) -> R.string.preference_allow_audio_video_autoplay
            getString(R.string.pref_key_block_autoplay_cellular_data) -> R.string.preference_block_autoplay_cellular_data
            getString(R.string.pref_key_block_audio_video_autoplay) -> R.string.preference_block_autoplay_audio_video
            else -> getString(R.string.preference_block_autoplay_audio_only)
        }
    }
```